### PR TITLE
Adds DMEtool to auto-sort/auto-merge DMEs

### PR DESCRIPTION
**Pull request by ghost**
I made a tool in lua that can auto-merge/auto-sort DMEs.
No CL due to no game-side changes

Requires Lua 5.1+ or LuaJIT 2.0+!

### CODE REVIEW

**Comment by deathride58** [comment]
This should try to replicate byond's sorting so that defines and proc overrides don't get completely fucked by the tool sorting things differently than how the DM compiler automatically sorts things.

There's a good reason why downstreams have to use separate folders that are listed lower in alphabetical order than the code folder for stuff to work.

File: tools/hooks/README.md [hunk 1]
```diff
@@ -13,7 +13,11 @@ automatically by GUI tools like TortoiseGit and GitHub for Windows, but
 
 ## Current Hooks
 
-* **Pre-commit**: Runs [mapmerge2] on changed maps, if any.
+* **Pre-commit**: Runs [mapmerge2] on changed maps, if any. Also sorts tgstation.dme via DMEtool (requires Lua 5.1+ or LuaJIT 2.0+!)
+ 
+## Current Merge Drivers
+
+* **dme**: runs DMEtool to auto merge the DMEs. Requirea Lua 5.1+ or LuaJIT 2.0+
```
Comment (SpaceManiac) [inline tools/hooks/README.md]: please

**Comment by lzimann** [comment]
Yes, unless the script matches exactly how Dream Maker does, it's pretty much useless. Also add a readme or something to actually explain how to use it.

File: tools/hooks/README.md (hunk 1 above)
Comment (MoreRobustThanYou) [reply tools/hooks/README.md]: ?

Comment (MoreRobustThanYou) [reply tools/hooks/README.md]: oh.

**Comment by KorPhaeron** [comment]
Keeps evading and triggering the stickyban on server and causing a big mess with false positives and his PRs are the only thing he might actually care enough about to get him to stop

So his PRs are closed

**Comment by KorPhaeron** [comment]
I've resolved the above issue, so

@tgstation/commit-access If anyone thinks this is worth having/is willing to review it then they are free to reopen it

**Pull request closed without merging**